src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -30,177 +44,512 @@ function NameTokenSelector({
+              <button
+                className="button flex flex-row center faded bold hover pointer"
+                style={{
+                  gap: '1em',
+                  border: '2px solid var(--text-faded)',
+                  borderRadius: '50px',
+                  height: '25px',
+                }}
+                onClick={() => {
+                  setSelectedToken(undefined);
+                  selectedTokenCallback(undefined);
+                }}
+              >
+                <CloseIcon
+                  width={'16px'}
+                  height={'16px'}
+                  fill={'var(--text-faded)'}
+                />
+                Remove
+              </button>
+            ) : (
+              <Tooltip
+                placement={'right'}
+                autoAdjustOverflow={true}
+                arrow={false}
+                overlayInnerStyle={{
+                  width: '190px',
+                  color: 'var(--text-black)',
+                  textAlign: 'center',
+                  fontFamily: 'Rubik-Bold',
+                  fontSize: '14px',
+                  backgroundColor: 'var(--text-white)',
+                  padding: '15px',
+                }}
+                title={
+                  'You can import an ANT by entering its contract ID, or search for one of your own by name, ticker, owner, or controller status, as well is its own contract ID'
+                }
+              >
+                Optional
+              </Tooltip>
+            )}
+          </span>
+        </div>
+        {/* selector dropdown */}
+        {tokens && searchActive ? (
+          <div
+            className="flex flex-column"
+            style={{
+              alignItems: 'flex-start',
+              justifyContent: 'flex-start',
+              gap: 0,
+              height: 'fit-content',
+              width: '100%',
+              backgroundColor: 'var(--card-bg)',
+              boxSizing: 'border-box',
+              zIndex: 10,
+            }}
+          >
+            {searchText && !filteredTokens?.length ? (
+              <span
+                className="text-large center"
+                style={{
+                  color: '#444547',
+                  margin: 'auto',
+                  height: '50px',
+                }}
+              >
+                No Results
+              </span>
+            ) : filteredTokens ? (
+              filteredTokens
+                .slice(
+                  Math.max((listPage - 1) * listItemCount, 0),
+                  listPage * listItemCount,
+                )
+                .map((token, index) => {
+                  if (!token) {
+                    return;
+                  }
+
+                  return (
+                    <button
+                      key={index}
+                      className="name-token-item pointer"
+                      onClick={() => {
+                        handleSetToken({
+                          id: token?.id,
```

Comment:
It seems this one is non-nullable by this point
```suggestion
                          id: token.id,
```

Reply:
fixed in https://github.com/ar-io/pdns-react/pull/229/commits/4de5520d092e8e98eaa9dcffe69150aac929361b